resources/views/docs/desktop/1/the-basics/notifications.md

```diff
@@ -40,8 +40,151 @@ Notification::title('Hello from NativePHP')
+
+```php
+Notification::title('Hello from NativePHP')
+    ->show();
+```
+
+### Notification Reference
+
+You can use the `reference()` method to set an event identifier and track which notification triggered a certain event. 
+This reference will be sent along with any event triggered by the notification. By default, a unique ID is generated as the reference.
+
+```php
+Notification::title('Hello from NativePHP')
+    ->reference(Str::uuid())
+    ->show();
+```
+
+### Notification Message
+
+You may set the notification's message using the `message()` method.
+
+```php
+Notification::title('Hello from NativePHP')
+    ->message('This is a detail message coming from your Laravel app.')
+    ->show();
+```
+
+### Notification Reply
+
+On macOS, you can allow the user to reply to a notification using the `hasReply()` method. 
+
+```php
+Notification::title('Hello from NativePHP')
+    ->hasReply()
+    ->show();
+```
+
+The `hasReply()` method accepts a placeholder reply message as an argument.
+
+```php
+Notification::title('Hello from NativePHP')
+    ->hasReply('This is a placeholder')
+    ->show();
+```
+
+### Notification Actions
+
+On macOS, you can add action buttons to a notification using the `addAction()` method. 
+
+```php
+Notification::title('Hello from NativePHP')
+    ->addAction('Click here')
+    ->show();
+```
+
+You can call the `addAction()` method multiple times if you need to add multiple buttons.
+
+```php
+Notification::title('Hello from NativePHP')
+    ->addAction('Button One')
+    ->addAction('Button Two')
+    ->show();
+```
+
 ## Events
 
 ### `NotificationClicked`
 The `Native\Laravel\Events\Notifications\NotificationClicked` event is dispatched when a user clicks on a notification.
 
+Example usage:
+```php
+Event::listen(NotificationClicked::class, function (NotificationClicked $event) {
+    $reference = $event->reference; // The unique reference to the clicked notification
+});
+```
+
+The reference can be used to track which notification was clicked:
+```php
+// Get recent posts
+$posts = Post::query()->where('created_at', '>', now()->subMinute())->get();
+
+// Generate notifications for recent posts
+$posts->each(function(Post $post) {
+    Notification::title('New post: ' . $post->title)
+                ->reference($post->id)
+                ->event(\App\Events\PostNotificationClicked::class)
+                ->show();
+});
+
+// Handle the click on a notification and redirect to the respective post
+Event::listen(\App\Events\PostNotificationClicked::class, function (\App\Events\PostNotificationClicked $event) {
+    $post = Post::findOrFail($event->reference);
+
+    Window::open()->url($post->url);
+});
+```
+
+### `NotificationClosed`
+The `Native\Laravel\Events\Notifications\NotificationClosed` event is dispatched when a user closes a notification.
+
+### `NotificationReply`
+The `Native\Laravel\Events\Notifications\NotificationReply` event is dispatched when a user replies to a notification.
```

Comment:
@simonhamp, do you think I should add an example of how to use the reply message? How generic would you like me to make the examples? I could do almost pseudo-code, just showing the idea or real-world examples.

Reply:
For now, no. This one feels limited to mac, so I think this is ok